components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,355 @@
+
+const AUTO_SHOW_CLASS = 'd2l-button-group-show';
+const AUTO_NO_SHOW_CLASS = 'd2l-button-group-no-show';
+
+const OPENER_TYPE = {
+	DEFAULT: 'default',
+	ICON: 'icon'
+};
+
+async function filterAsync(arr, callback) {
+	const fail = Symbol();
+	const results = await Promise.all(arr.map(async item => {
+		const callbackResult = await callback(item);
+		return callbackResult ? item : fail;
+	}));
+	return results.filter(i => i !== fail);
+}
+
+export const OverflowGroupMixin = superclass => class extends LocalizeCoreElement(superclass) {
+
+	static get properties() {
+		return {
+			/**
+			 * Use predefined classes on slot elements to set min and max buttons to show
+			 * @type {boolean}
+			 */
+			autoShow: {
+				type: Boolean,
+				attribute: 'auto-show',
+			},
+			/**
+			 * @ignore
+			 */
+			chompIndex: {
+				state: true
+			},
+			/**
+			 * minimum amount of buttons to show
+			 * @type {number}
+			 */
+			minToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'min-to-show',
+			},
+			/**
+			 * maximum amount of buttons to show
+			 * @type {number}
+			 */
+			maxToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-to-show',
+			},
+			/**
+			 * @ignore
+			 */
+			mini: {
+				state: true
+			},
+			/**
+			 * @ignore
+			 */
+			openerType: {
+				type: String,
+				attribute: 'opener-type'
+			}
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-overflow-group-container {
+				display: flex;
+				flex-wrap: wrap;
+				justify-content: var(--d2l-overflow-group-justify-content, normal);
+			}
+			.d2l-overflow-group-container ::slotted([data-is-chomped]) {
+				display: none !important;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this._handleItemMutation = this._handleItemMutation.bind(this);
+		this._handleResize = this._handleResize.bind(this);
+
+		this._throttledResize = (entries) => requestAnimationFrame(() => this._handleResize(entries));
+
+		this._overflowHidden = false;
+		this.autoShow = false;
+		this.maxToShow = -1;
+		this.minToShow = 1;
+		this.mini = this.openerType === OPENER_TYPE.ICON;
+		this.openerType = OPENER_TYPE.DEFAULT;
+		this._resizeObserver = null;
+		this._slotItems = [];
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+
+		requestAnimationFrame(async() => {
+			await this.updateComplete;
+
+			const container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+			this._resizeObserver = new ResizeObserver(this._throttledResize);
+			this._resizeObserver.observe(container);
+		});
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
+	}
+
+	async firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this._buttonSlot = this.shadowRoot.querySelector('slot');
```

Comment:
This doesn't seem very expensive, so I'd be tempted to just move this down to the one place it's used inside `_getSlotItems`, which would remove the need for `firstUpdated` and remove a member variable. *shrug*